src/priceClient/adapters/coingecko.ts

```diff
@@ -0,0 +1,68 @@
+import assert from "assert";
+import axios, { AxiosError } from "axios";
+import { PriceFeedAdapter, TokenPrice } from "../priceClient";
+
+type CoinGeckoTokenPrice = {
+  [currency: string]: number;
+  last_updated_at: number;
+};
+
+type CoinGeckoPriceResponse = {
+  [address: string]: CoinGeckoTokenPrice;
+};
+
+const defaultTimeout = 5000; // mS
+
+export class PriceFeed implements PriceFeedAdapter {
+  public readonly host: string;
+  protected _timeout = defaultTimeout;
+
+  constructor(public readonly name: string, private readonly apiKey?: string) {
+    this.host = this.apiKey ? "pro-api.coingecko.com" : "api.coingecko.com";
+  }
+
+  get timeout(): number {
+    return this._timeout;
+  }
+
+  set timeout(timeout: number) {
+    assert(timeout >= 0);
+    this._timeout = timeout;
+  }
+
+  async getPriceByAddress(address: string, currency = "usd"): Promise<TokenPrice> {
+    const price: TokenPrice[] = await this.getPricesByAddress([address], currency);
+    return price[0];
+  }
+
+  async getPricesByAddress(addresses: string[], currency = "usd"): Promise<TokenPrice[]> {
+    const prices: CoinGeckoPriceResponse = await this.query(addresses, currency);
+    return addresses.map((addr: string) => {
+      const price: CoinGeckoTokenPrice = prices[addr.toLowerCase()];
+      return { address: addr, price: price[currency], timestamp: price.last_updated_at };
+    });
+  }
+
+  private async query(
+    addresses: string[],
+    baseCurrency = "usd",
+    timeout: number = defaultTimeout
```

Comment:
Does this function not respect the configured timeout in the class?

Reply:
This has now been consolidated as part of this commit: https://github.com/across-protocol/sdk-v2/pull/98/commits/fb5780a3c40637f17247612e7bbfc988319d3b9a

I'd probably like to have the theoretical ability to override this per-call, but we don't have a need for it right now.